lib/hdm/client/smart_client.rb

```diff
@@ -102,6 +111,19 @@ def discover_fhir_endpoint_params
         options = fhir_client.get_oauth2_metadata_from_conformance
         options
       end
+
+      def get_auth_params(params = {})
+        params[:scope] = generate_scopes_from_conformance unless provider.scopes
+        super(params)
+      end
+
+      def generate_scopes_from_conformance
+        scopes = ''
+        client_capability_statement.rest[0].resource.each do |r|
```

Comment:
Just as devil's advocate, this could also be done with `.map` and `.join` . But I think this is clearer so don't change it